block/test/BlockSpec.hs

```diff
@@ -0,0 +1,67 @@
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE RecordWildCards #-}
+{-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE NoImplicitPrelude #-}
+{-# LANGUAGE ConstraintKinds #-}
+{-# LANGUAGE TypeSynonymInstances #-}
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE FlexibleContexts     #-}
+{-# OPTIONS_GHC -Weverything -Wno-unsafe -Wno-type-defaults #-}
```

Comment:
I'd be curious as to what needs the `-Wno-type-defaults`, because it isn't obvious from the code.

Reply:
i make a habit of turning on `-Weverything` so i can see any minor issue i'm causing